### PR TITLE
Add --config option

### PR DESCRIPTION
**Pull request by goerz**
This addresses https://github.com/myint/rstcheck/issues/56

This is a tentative implementation of my proposed solution of adding a `--config` option, open to discussion. This pull request is not yet ready to be merged; at the very least, it would need additional documentation and tests.

### CODE REVIEW

File: rstcheck.py [hunk 1]
```diff
@@ -471,18 +471,26 @@ def ignore_sphinx():
...
+            print('using config file %s' % directory_or_file)
+        return directory_or_file
+    directory = directory_or_file
 
     while directory:
         for filename in CONFIG_FILES:
             candidate = os.path.join(directory, filename)
             if os.path.exists(candidate):
+                if debug:
+                    print('using config file %s' % candidate)
```
Comment (myint) [inline rstcheck.py]: Please switch this to:

```python
print('Using config file {}'.format(candidate),
      file=sys.stderr)
```

**Comment by myint** [comment]
Seems reasonable to me! I'll take another look once you complete the tests/documentation you mention.

File: rstcheck.py (hunk 1 above)
Comment (goerz) [reply rstcheck.py]: force-pushed to fix this

**Comment by goerz** [comment]
Documentation and tests have been added in a second commit. I also documented the previously undocumented possibility to have the configuration in `setup.cfg`. I took the liberty of switching the order of `docs` and `foo` in the example file structure in the documentation, as that's the structure a Python project `foo` would generally have. When talking about `setup.cfg`, it seemed  confusing to have a project `docs` with a documentation folder `foo`, as opposed to a project `foo` with a documentation folder `docs`.

This should now be ready to merge if @myint  has no further objections.

File: README.rst [hunk 2]
```diff
@@ -183,7 +185,29 @@ For example, consider a project with the following directory structure::
...
+``rstcheck`` inside the general ``setup.cfg`` `distutils configuration file`_,
+in the project root.
+
+You can override the location of the config file with the ``--config`` argument::
+
+    $ rstcheck --config $HOME/.rstcheck.ini foo/docs/bar.rst
+
+will use the file ``.rstcheck.ini`` in your home directory. If the argument to
+``--config`` is a directory, ``rstcheck`` will search that directory and any
+any of its ancestors for a file ``.rstcheck.cfg`` or ``setup.cfg``::
```
Comment (stsewd) [inline README.rst]: `any` is duplicated

File: README.rst [hunk 3]
```diff
@@ -146,13 +148,13 @@ Configuration file
...
+directory or ancestor directories of the file it is checking.
 
 For example, consider a project with the following directory structure::
 
-    docs
-    ├── foo
+    foo
+    ├── docs
     │   └── bar.rst
     ├── index.rst
```
Comment (stsewd) [inline README.rst]: `index.rst` should be at the same level of `bar.rst` now

**Comment by goerz** [comment]
Thanks @stsewd  for pointing these out. So I think this is merged now, so I don't know if I can still fix these typos on my branch and amend this pull request. @myint, do you just want to make these two fixes on your own master? That might be easier than working through a pull request.

**Comment by myint** [comment]
Please go ahead and create a pull request. Thanks!

**Pull request merged**